pkg/controller/volume/persistentvolume/pv_controller_base.go

```diff
@@ -309,6 +309,83 @@ func (ctrl *PersistentVolumeController) Run(stopCh <-chan struct{}) {
 	<-stopCh
 }
 
+func (ctrl *PersistentVolumeController) updateClaimMigrationAnnotations(claim *v1.PersistentVolumeClaim) (*v1.PersistentVolumeClaim, error) {
+	// TODO: update[Claim|Volume]MigrationAnnotations can be optimized to not
+	// copy the claim/volume if no modifications are required. Though this
+	// requires some refactoring as well as an interesting change in the
+	// semantics of the function which may be undesirable. If no copy is made
+	// when no modifications are required this function could sometimes return a
+	// copy of the volume and sometimes return a ref to the original
+	copiedClaim := claim.DeepCopy()
+	modified := annealMigrationAnnotations(ctrl.csiMigratedPluginManager, ctrl.translator, copiedClaim.Annotations, pvutil.AnnStorageProvisioner)
+	if modified {
```

Comment:
Even though it's space shuttle code, I think we're allowed to invert this condition and have an early return.